### PR TITLE
active mode nav + open active tab on mount

### LINKED ISSUE #109: Active mode set on content nav when active
The navigation menu should reflect which bit of content is currently active. It should also persist the accordion state for whichever content item is currently being viewed, i.e. the navigation should stay open on that lesson if the navigation is opened after navigating.
As it is now, once you close the navigation it also collapses all accordions.

### PR DESCRIPTION
**Pull request by StianSto**
Added a feature to highlight the current document. for now it matches the exact path. I chose a pretty random color. im sure we'll come back to a design revamp sooner or later anyway :) this resolves #109

### CODE REVIEW

**Review by Melisa-Zorraindo — APPROVED** [review]
I tried to find a solution to stay in theme with the colour, but because of how the hover effect is implemented, the .active class has to be applied to the .entity class. So I guess the hardcoded colour will have to do for now. Plus, it looks better than any other in the theme.

**Comment by Fermain** [comment]
I'm not seeing any highlights - am I missing something?

File: packages/lms/src/lib/components/navigation/Entity.svelte [hunk 1]
```diff
@@ -31,7 +40,11 @@
   <div class="entity-inner">
     <div class="entity-header hover:bg-primary-hover-token rounded-container-token p-2">
       {#if entity.browserPath}
-        <a href={entity.browserPath} on:click={drawerClose} class="entity-title flex flex-row items-center">
+        <a
+          href={entity.browserPath}
+          on:click={drawerClose}
```
Comment (Fermain) [inline packages/lms/src/lib/components/navigation/Entity.svelte:45]:
```suggestion
          on:click={drawerClose}
          class:active={active}
```

Not tested but this is a more sveltey way

Comment (pretzL) [reply packages/lms/src/lib/components/navigation/Entity.svelte:45]: You can just do `class:active` to be even more Sveltey

**Comment by Melisa-Zorraindo** [comment]
I saw the lesson highlighted, not the course, when I tried it yesterday

**Comment by Fermain** [comment]
It's an issue with onMount - you have to refresh to re-render the component

**Comment by Melisa-Zorraindo** [comment]
Where there any changes made after I approved? I didn't have to refresh to see the active link

**Comment by Fermain** [comment]
> Where there any changes made after I approved? I didn't have to refresh to see the active link

Nope, pretty sure I tested the same thing as you

**Comment by Melisa-Zorraindo** [comment]
Maybe I was clicking on the exact same lesson every time I was trying to find a way to get the theme variables to work, sorry

**Comment by pretzL** [comment]
I'm seeing the same as Melisa. I don't have to do anything extra to see the course highlighted like reloading the component.

**Comment by Fermain** [comment]
You aren't allowed to gang up on me

**Comment by pretzL** [comment]
> You aren't allowed to gang up on me

Consider yourself ganged up on

**Comment by Fermain** [comment]
> Consider yourself ganged up on

gung upon

**Comment by pretzL** [comment]
> > Consider yourself ganged up on
> 
> gung upon

gungst've

**Comment by Fermain** [comment]
I tested again with a full rebuild and I have the same behaviour

**Comment by pretzL** [comment]
Can confirm it's a desktop vs mobile issue. Mobile works fine, desktop does not.
Alternative solution to the `onMount` issue is to custom define reactivity:
```svelte
let currentPath = "";
onMount(() => {
  currentPath = decodeURI(window.location.pathname);
});

$: {
  if (currentPath + '/' === entity.browserPath) active = true;
  if (currentPath.includes(entity.browserPath)) toggle();
}
  ```

**Review by pretzL — APPROVED** [review]
This works well on both desktop and mobile now from my testing. Good work.

**Pull request merged**